<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1600</width>
    <height>900</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="la_origin">
    <property name="geometry">
     <rect>
      <x>1200</x>
      <y>60</y>
      <width>380</width>
      <height>190</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>原图显示区域</string>
    </property>
    <property name="text">
     <string>原图</string>
    </property>
   </widget>
   <widget class="QLabel" name="la_canny">
    <property name="geometry">
     <rect>
      <x>1200</x>
      <y>320</y>
      <width>380</width>
      <height>190</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Canny图显示区域</string>
    </property>
    <property name="text">
     <string>Canny图</string>
    </property>
   </widget>
   <widget class="QLabel" name="la_mask">
    <property name="geometry">
     <rect>
      <x>1200</x>
      <y>580</y>
      <width>380</width>
      <height>190</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Mask图显示区域</string>
    </property>
    <property name="text">
     <string>Mask图</string>
    </property>
   </widget>
   <widget class="QScrollArea" name="scrollArea">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>60</y>
      <width>1111</width>
      <height>711</height>
     </rect>
    </property>
    <property name="widgetResizable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="scrollAreaWidgetContents">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>1109</width>
       <height>709</height>
      </rect>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="la_threshold1">
    <property name="geometry">
     <rect>
      <x>1300</x>
      <y>0</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>threshhold1:</string>
    </property>
   </widget>
   <widget class="QLabel" name="la_threshold2">
    <property name="geometry">
     <rect>
      <x>1450</x>
      <y>10</y>
      <width>91</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>threshhold2:</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="sb_threshold1">
    <property name="geometry">
     <rect>
      <x>1390</x>
      <y>10</y>
      <width>48</width>
      <height>26</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Canny函数的threshhold1</string>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>500</number>
    </property>
    <property name="value">
     <number>50</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="sb_threshold2">
    <property name="geometry">
     <rect>
      <x>1540</x>
      <y>10</y>
      <width>48</width>
      <height>26</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Canny函数的threshhold2</string>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>500</number>
    </property>
    <property name="value">
     <number>100</number>
    </property>
   </widget>
   <widget class="QPushButton" name="pb_open">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>10</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>打开图片列表文件</string>
    </property>
    <property name="text">
     <string>打开</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pb_pre">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>10</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>A, 打开下一张</string>
    </property>
    <property name="text">
     <string>上一张</string>
    </property>
    <property name="shortcut">
     <string>A</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pb_next">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>10</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>D, 打开下一张</string>
    </property>
    <property name="text">
     <string>下一张</string>
    </property>
    <property name="shortcut">
     <string>D</string>
    </property>
   </widget>
   <widget class="QLabel" name="la_blur">
    <property name="geometry">
     <rect>
      <x>1200</x>
      <y>0</y>
      <width>41</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>blur:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="cb_blur">
    <property name="geometry">
     <rect>
      <x>1240</x>
      <y>10</y>
      <width>41</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>Blur函数的ksize大小</string>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <item>
     <property name="text">
      <string>1</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>3</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>5</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>7</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>9</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>11</string>
     </property>
    </item>
   </widget>
   <widget class="QCheckBox" name="cb_origin">
    <property name="geometry">
     <rect>
      <x>1300</x>
      <y>810</y>
      <width>71</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>显示原图</string>
    </property>
    <property name="text">
     <string>原图</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="cb_canny">
    <property name="geometry">
     <rect>
      <x>1380</x>
      <y>810</y>
      <width>92</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>显示Canny图</string>
    </property>
    <property name="text">
     <string>Canny图</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="cb_mask">
    <property name="geometry">
     <rect>
      <x>1480</x>
      <y>810</y>
      <width>92</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>显示Mask图</string>
    </property>
    <property name="text">
     <string>Mask图</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="cb_origin_canny">
    <property name="geometry">
     <rect>
      <x>1180</x>
      <y>810</y>
      <width>101</width>
      <height>23</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>显示原图+Canny图</string>
    </property>
    <property name="text">
     <string>原图+Canny</string>
    </property>
   </widget>
   <widget class="QLabel" name="la_scale">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>0</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>放大:</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="sb_scale">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>10</y>
      <width>48</width>
      <height>26</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>放大倍数</string>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>10</number>
    </property>
    <property name="value">
     <number>1</number>
    </property>
   </widget>
   <widget class="QSpinBox" name="sb_pensize">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>10</y>
      <width>48</width>
      <height>26</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>W, 笔的大小</string>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
    <property name="value">
     <number>5</number>
    </property>
   </widget>
   <widget class="QLabel" name="la_pensize">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>0</y>
      <width>51</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>笔宽:</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pb_cancel">
    <property name="geometry">
     <rect>
      <x>380</x>
      <y>10</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>F, 撤销上一次操作</string>
    </property>
    <property name="text">
     <string>撤销</string>
    </property>
    <property name="shortcut">
     <string>F</string>
    </property>
   </widget>
   <widget class="QLabel" name="la_mode">
    <property name="geometry">
     <rect>
      <x>850</x>
      <y>10</y>
      <width>51</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>模式:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="cb_mode">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>10</y>
      <width>61</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>E, 选择绘制模式</string>
    </property>
    <item>
     <property name="text">
      <string extracomment="覆盖检测出的Canny边">涂抹</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string extracomment="手动补上没检测出来的Canny边">标记</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string extracomment="消除不想标的阴影边">消除</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="pb_save">
    <property name="geometry">
     <rect>
      <x>880</x>
      <y>810</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>保存标记好的Mask和去除不想要区域的原图</string>
    </property>
    <property name="text">
     <string>保存</string>
    </property>
    <property name="shortcut">
     <string>S</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="le_save">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>810</y>
      <width>711</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>保存路径</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="placeholderText">
     <string>保存路径...</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pb_choose">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>810</y>
      <width>31</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>选择保存位置</string>
    </property>
    <property name="text">
     <string>...</string>
    </property>
   </widget>
   <widget class="QLabel" name="la_canny_mode">
    <property name="geometry">
     <rect>
      <x>980</x>
      <y>10</y>
      <width>67</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Canny源:</string>
    </property>
   </widget>
   <widget class="QComboBox" name="cb_canny_mode">
    <property name="geometry">
     <rect>
      <x>1050</x>
      <y>10</y>
      <width>91</width>
      <height>25</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>灰度图</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>亮度图</string>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="pb_fullscreen">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>10</y>
      <width>89</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string>R, 放大绘制控件</string>
    </property>
    <property name="text">
     <string>扩展</string>
    </property>
    <property name="shortcut">
     <string>R</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1600</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
